perses/annihilation/relative.py

```diff
@@ -2692,39 +2654,197 @@ def __init__(self,
         # Generate the topology representation
         self._hybrid_topology = self._create_topology()
 
-        # define the scale and nonscale region
-        self._make_scale_regions(scale_region)
+    def _handle_scale_regions(self, scale_regions):
+        num_scale_regions = None
+        if scale_regions is None:
+            self._num_scale_regions = 0
+            self._scale_regions = None
+        else:
+            assert type(scale_regions) == list
+            num_scale_regions = len(scale_regions)
+            for entry in scale_regions:
+                assert type(entry) == list
+
+            #the last bit is to ensure that the scale regions are disjoint
+            num_scaled_particles = sum([len(i) for i in scale_regions])
+            import itertools
+            num_combined_scaled_particles = len(list(itertools.chain.from_iterable(scale_regions)))
+            assert num_scaled_particles == num_combined_scaled_particles, f"each scale region must be disjoint"
+            self._scale_regions = [set(i) for i in scale_regions]
+
```

Comment:
note that you never set `self._num_scale_regions` in this else statement. 
I would recommend not having both properties `self._num_scale_regions` and `self._scale_regions`. You can just use `len(self._scale_regions)` to retrieve the `num_scale_regions`. Or, you could define a helper function `get_num_scale_regions()` that retrieves the number of scale regions based on `self._scale_regions`. It's not ideal to have to update two different properties related to the same variable every time the variable changes.